rasa/core/policies/intent_ted_policy.py

```diff
@@ -0,0 +1,693 @@
+            )
+
+            # If score for query intent is below threshold and
+            # the query intent is not the top likely intent
+            if (
+                query_intent_similarity < self.label_thresholds[query_intent_id]
+                and query_intent_id != sorted_intent_scores[-1][0]
+            ):
+                logger.debug(
+                    f"Intent {query_intent}-{query_intent_id} unlikely to occur here."
+                )
+                return True
+
+        return False
+
+    def persist_model_utilities(self, model_path: Path) -> None:
+        """Persist model's utility attributes like model weights, etc.
+
+        Args:
+            model_path: Path where model is to be persisted
+        """
+        super().persist_model_utilities(model_path)
+        io_utils.pickle_dump(
+            model_path / f"{self._metadata_filename()}.label_thresholds.pkl",
+            self.label_thresholds,
+        )
+
+    @classmethod
+    def _load_model_utilities(cls, model_path: Path) -> None:
+        """Load model's utility attributes.
+
+        Args:
+            model_path: Path where model is to be persisted.
+        """
+        model_utilties = super()._load_model_utilities(model_path)
+        label_thresholds = io_utils.pickle_load(
+            model_path / f"{cls._metadata_filename()}.label_thresholds.pkl"
+        )
+        model_utilties.update({"label_thresholds": label_thresholds})
+
+    @classmethod
+    def _update_loaded_params(cls, meta: Dict[Text, Any]) -> Dict[Text, Any]:
+        meta = train_utils.override_defaults(cls.defaults, meta)
+
+        return meta
+
+    @classmethod
+    def _load_policy_with_model(
+        cls,
+        model: "IntentTED",
+        featurizer: TrackerFeaturizer,
+        model_utilities: Dict[Text, Any],
+        should_finetune: bool,
+    ) -> "IntentTEDPolicy":
+        return cls(
+            featurizer=featurizer,
+            priority=model_utilities["priority"],
+            model=model,
+            fake_features=model_utilities["fake_features"],
+            entity_tag_specs=model_utilities["entity_tag_specs"],
+            should_finetune=should_finetune,
+            label_thresholds=model_utilities["label_thresholds"],
+            **model_utilities["meta"],
+        )
+
+
+class IntentTED(TED):
+    def _prepare_label_classification_layers(self, predictor_attribute: Text) -> None:
+        """Prepares layers & loss for the final label prediction step."""
+        self._prepare_embed_layers(predictor_attribute)
+        self._prepare_embed_layers(LABEL)
+
+        self._prepare_dot_product_loss(
+            LABEL, self.config[SCALE_LOSS], loss_layer=layers.MultiLabelDotProductLoss
+        )
+
+    def _prepare_dot_product_loss(
+        self,
+        name: Text,
+        scale_loss: bool,
+        prefix: Text = "loss",
+        loss_layer: tf.keras.layers.Layer = layers.SingleLabelDotProductLoss,
```

Comment:
`MultiLabelDotProductLoss` is being passed as the argument value in `_prepare_label_classification_layers`. Needed the default value to match the function signature with its counterpart in `TED`.

Reply:
But you could use a class variable `self.DEFAULT_LOSS` or something and use that as default, no? Would be cleaner since IntentTED really should use the MultiLabel loss.